src/modules/aws_vpc/mappers/route.ts

```diff
@@ -0,0 +1,217 @@
+import _ from 'lodash';
+
+import { EC2 } from '@aws-sdk/client-ec2';
+import { Route as AwsRoute } from '@aws-sdk/client-ec2/dist-types/models/models_1';
+
+import { AWS, crudBuilder2 } from '../../../services/aws_macros';
+import { Context, Crud2, IdFields, MapperBase } from '../../interfaces';
+import { Route, RouteTable } from '../entity';
+import { AwsVpcModule } from '../index';
+
+export class RouteMapper extends MapperBase<Route> {
+  module: AwsVpcModule;
+  entity = Route;
+  entityId = (e: Route) => `${e.routeTable.routeTableId ?? e.routeTable.id}|${e.destination}|${e.region}`;
+  idFields = (id: string) => {
+    const [routeTableId, destination, region] = id.split('|');
+    return { routeTableId, destination, region };
+  };
+  generateId = (fields: IdFields) => {
+    const requiredFields = ['routeTableId', 'destination', 'region'];
+    if (
+      Object.keys(fields).length !== requiredFields.length &&
+      !Object.keys(fields).every(fk => requiredFields.includes(fk))
+    ) {
+      throw new Error(`Id generation error. Valid fields to generate id are: ${requiredFields.join(', ')}`);
+    }
+    return `${fields.routeTableId}|${fields.destination}|${fields.region}`;
+  };
+  equals = (a: Route, b: Route) => {
+    return this.eqRoute(a, b);
+  };
+  cidrIPv4Pattern = new RegExp(
+    '^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?).){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/(3[0-2]|[12][0-9]|[0-9])$',
+  );
+  cidrIPv6Pattern = new RegExp(
+    '^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:))/(12[0-8]|1[01][0-9]|[1-9]?[0-9])$',
+  );
+  prefixedListPattern = new RegExp('^pl-[wd]+$');
```

Comment:
Nit: This check is slightly stricter than the postgres check, that could cause records that pass in the DB but fail in the engine